org-code-javabuilder/lib/src/main/java/org/code/neighborhood/Painter.java

```diff
@@ -0,0 +1,144 @@
+package org.code.neighborhood;
+
+public class Painter {
+    private int xLocation;
+    private int yLocation;
+    private Direction direction;
+    private int remainingPaint;
+    private Grid grid;
+    private String id;
+
+    public Painter(int x, int y, String direction, int paint) {
+        this.xLocation = x;
+        this.yLocation = y;
+        this.direction = new Direction(direction);
+        this.remainingPaint = paint;
+        this.grid = Grid.getInstance();
+        this.id = "Painter-"+this.grid.registerPainter();
+
+        System.out.println("created painter");
+    }
+
+    // Turn the painter one compass direction left (i.e. North -> West)
+    public void turnLeft() {
+        this.direction.turnLeft();
```

Comment:
If we make direction itself an enum, this line would be something like `this.direction = this.direction.turnLeft()`.